node/network/approval-distribution/src/lib.rs

```diff
@@ -63,11 +68,15 @@ const COST_DUPLICATE_MESSAGE: Rep = Rep::CostMinorRepeated("Peer sent identical
 const COST_ASSIGNMENT_TOO_FAR_IN_THE_FUTURE: Rep =
 	Rep::CostMinor("The vote was valid but too far in the future");
 const COST_INVALID_MESSAGE: Rep = Rep::CostMajor("The vote was bad");
+const COST_OVERSIZED_BITFIELD: Rep = Rep::CostMajor("Oversized certificate or candidate bitfield");
 
 const BENEFIT_VALID_MESSAGE: Rep = Rep::BenefitMinor("Peer sent a valid message");
 const BENEFIT_VALID_MESSAGE_FIRST: Rep =
 	Rep::BenefitMinorFirst("Valid message with new information");
 
+// Maximum valid size for the `CandidateBitfield` in the assignment messages.
+const MAX_BITFIELD_SIZE: usize = 500;
```

Comment:
Is this a hardcoded limit on the number of cores? What was the rationale of doing that vs looking into runtime? I don't expect using runtime values to be much of performance hit if we assume number of cores doesn't change within a session and cache it.

Reply:
The reasoning is that we want to avoid to also fetch session info in approval distribution, I know it will await only for first query per relay parent + on new session, but the 2 extra cache lookups will add overhead to all messages before any book keeping.

500 cores should be reasonable max size assuming we're gonna be doing Execution(Core) Groups by the time we get to this usage/scale.